### ISSUE TITLE
FPS is always limited

### ISSUE THREAD

**Issue by Korijn** [pygfx/wgpu-py#256]
For some reason it's impossible to push FPS over screen refresh rate (commonly 30 or 60).

See pygfx/pygfx#278 and pygfx/pygfx#254

**Comment by almarklein** [comment]
Wow!

Ok, so I just put my charger in, and tried again. And to my surprise, I get 100+ FPS on both examples on Win 11 now (both for glfw and Qt). I can even plug/unplug my laptop while an example is running and see the FPS switch between 30 and 100+ ...

So it looks like Win11 clips the framerate at the OS level when on battery power. Can someone check on Win10?

**Comment by Korijn** [comment]
Win11 indeed switches the refresh rate based on power availability.

But it still sounds like it's limited to not go over display refresh rate in both cases. With power your refresh rate could be as high as 120 or 144.

I think the question here is; can we render unrestricted just to measure how far an example can be pushed?

**Comment by almarklein** [comment]
Ah, you mean that even with power, the refresh rate is still the limit, even though it's now higher ...

**Comment by Korijn** [comment]
Exactly.

**Comment by Korijn** [comment]
Just tested with glfw; still limited to 60 fps.

**Comment by almarklein** [comment]
I created a script to test this:
```py
import time

# import PySide6
from wgpu.gui.glfw import WgpuCanvas, run
import wgpu.backends.rs


# Create canvas
canvas = WgpuCanvas(max_fps=1000000)


# Setup wgpu
adapter = wgpu.request_adapter(canvas=canvas, power_preference="high-performance")
device = adapter.request_device()
present_context = canvas.get_context()
render_texture_format = present_context.get_preferred_format(device.adapter)
present_context.configure(device=device, format=render_texture_format)


FPS = {"t": time.perf_counter(), "n": 1}


def animate():
    canvas.request_draw()

    # Show FPS
    now = time.perf_counter()
    if now - FPS["t"] < 1:
        FPS["n"] += 1
    else:
        fps = FPS["n"] / (now - FPS["t"])
        print(f"FPS: {fps:0.1f}")
        FPS["t"] = now
        FPS["n"]= 1

    # Actually draw.
    # Disabling the outer IF gives an FPS of 40k FPS
    # Disabling only the inner IF gives an FPS of ~ 16K FPS
    # Enabling both IFs gives ~ 110 FPS
    if True:

        command_encoder = device.create_command_encoder()

        if True:
            current_texture_view = present_context.get_current_texture()
            render_pass = command_encoder.begin_render_pass(
                color_attachments=[
                    {
                        "view": current_texture_view,
                        "resolve_target": None,
                        "clear_value": (0.0, 0.8, 0.2, 1),
                        "load_op": wgpu.LoadOp.clear,
                        "store_op": wgpu.StoreOp.store,
                    }
                ],
            )
            render_pass.end()

        device.queue.submit([command_encoder.finish()])


canvas.request_draw(animate)
run()
```

**Comment by almarklein** [comment]
What I found so far is that renderpass apparently waits for a vsync. Then a bell rang in my distant memory, and indeed there is [the present_mode](https://github.com/gfx-rs/wgpu/blob/e54a36ee78693aca0120458bcc2c03262a715f1e/wgpu-types/src/lib.rs#L2663-L2678), that we set (hard-coded) here:

https://github.com/pygfx/wgpu-py/blob/ce16ad60af474c085fc9502417462d87c7ab19f7/wgpu/backends/rs.py#L373-L383 

In short, we always tell wgpu to wait for vsync.

Setting this to 0 indeed bumps up the FPS, but only by a bit, sticking to just under 200 FPS, which seems rather low for presenting an empty frame. Maybe something else is at play? This is on MacOS, will try on Win11 next.

**Comment by almarklein** [comment]
Yes, but from what I've seen now, it's not that really a solution. It looks like Windows ignores it when on battery, and on MacOS it seems like there is another limit at play.

**Comment by Korijn** [comment]
Well, I could be wrong here, but I expect the vast majority of graphics programmers who are interested in this type of functionality are benching on Windows machines, either plugged in or desktop.
